scripts/autobahn-client.sh

```diff
@@ -23,10 +18,12 @@ function test_diff() {
     fi
 }
 
-cargo build --release --example autobahn-client
+docker run -d --rm \
+    -v "${PWD}/autobahn:/autobahn" \
+    -p 9001:9001 \
+    crossbario/autobahn-testsuite \
+    wstest -m fuzzingserver -s 'autobahn/fuzzingserver.json'
```

Comment:
We start the container in the background (using `-d`), but AFAICS we never tear down the container.

Instead of removing the `function cleanup()` and `trap`, we should update them to tear down the container. `docker run -d` returns container ID, which can be assigned to a variable for that use-case.

Reply:
Ok, that's a valid concern that I had, but I decided to not do it, because I assumed that once the job ends, so do all the processes, including docker and its containers. Because even if I stop the container (just for the sake of not having some containers running at the end of the job), it is still be there (it won't be pruned automatically).

> Applies also to scripts/autobahn-server.sh.

Does it? I'm asking because in case of `autobahn-server` we start our own server (that is covered by the `cleanup()`) and we use Autobahn only for the fuzzing client. Once the client process ends, the container is stopped. The same for

Reply:
> Ok, that's a valid concern that I had, but I decided to not do it, because I assumed that once the job ends, so do all the processes, including docker and its containers.

I was thinking not just about the CI, but more about people that would run that locally (to test tungstenite changes/to test new autobah/you testing this PR). And just as a general good practice to clean up resources that are created.

> Because even if I stop the container (just for the sake of not having some containers running at the end of the job), it is still be there (it won't be pruned automatically).

This looks like a problem, because we specify `--rm`, so the container should be destroyed. Let me guess: the container suffers from the "PID 1" problem, i.e. the CTRL+C signal is not caught by it properly, so that it is not stopped right away after `docker stop` and is killed only after the graceful stop timeout (10s by default). If this is the case, mitigation is easy: add `--init` to `docker run` flags.

(reply to the other part in other code comment)

Reply:
> If this is the case, mitigation is easy: add --init to docker run flags.

Oh nice, thanks for the info, I was afraid that the whole `dumb-init` thing would be required.